src/LanguageServer/Impl/LanguageServer.cs

```diff
@@ -452,6 +460,26 @@ public Task LoadExtension(JToken token, CancellationToken cancellationToken)
             _searchPaths = _initParams.initializationOptions.searchPaths;
         }
 
+        private async Task LogOnException(Func<Task> func) {
```

Comment:
We should try sending telemetry via `ITelemetryServce`. We can craft test event and check how it shows up in the app insights

Reply:
Yes, I know. Above I stated that I specifically didn't put it in this WIP because I hadn't looked into the right schema for the telemetry events (Since ITelemetryService seems to just accept `object`).